### ISSUE TITLE
Bug in Recognizing Split Bin Files During Model Conversion Process

### ISSUE THREAD

**Issue by Aintor** [MochiDiffusion/MochiDiffusion#437]
### Processor

M1 Pro (or later)

### Memory

16GB

### What happened?

When I try to convert a diffuser model to an MLMODELC model, it shows an error:
`OSError: Error no file named diffusion_pytorch_model.bin found in directory PVC_diffusers/unet.`
I found that the bin file was split into two parts: `diffusion_pytorch_model-00001-of-00002.bin` and `diffusion_pytorch_model-00002-of-00002.bin`.
<img width="459" alt="the file" src="https://github.com/MochiDiffusion/MochiDiffusion/assets/86539564/787d477f-bf0e-4445-b2e8-bd53e549c90c">
The conversion code should be modified to handle this situation.

**Comment by rayenyang** [comment]
have the same error

**Comment by danni-cool** [comment]
> check out #441

how to apply this patch to ml-stable-diffusion?

**Comment by danni-cool** [comment]
same issue

**Comment by Aintor** [comment]
> > check out #441
> 
> 
> 
> how to apply this patch to ml-stable-diffusion?

Once the pr is approved to merge into the main branch by the repo keeper, the change will be applied. Or you can modify your repo by the pr so you can fix the bugs by yourself earlier.

**Comment by danni-cool** [comment]
> > > check out #441
> > 
> > 
> > how to apply this patch to ml-stable-diffusion?
> 
> Once the pr is approved to merge into the main branch by the repo maintainer, the change will be applied. Or you can modify your repo with the pr so you can fix the bugs yourself earlier.

but are we discussing "model convert breaking of split bin files"? this pr is based on "apple/ml-stable-diffusion", not mochi-diffusion

**Comment by Aintor** [comment]
> but are we discussing "sd model convert breaking of split bin files"? this pr is based on "apple/ml-stable-diffusion", not mochi-diffusion

You can try replacing the files modified in the PR with those in your repo and then test to see if it works, or contact the PR holder for further help. I think this thread is mainly for bug reports rather than for solving issues.

**Comment by danni-cool** [comment]
thanks, I'll try

**Comment by danni-cool** [comment]
@vlad-ivanov-name  Thanks for your generous answer and effort in solving this problem. This gave a guidance to continue my conversion workflow. and finally, I succeeded!

Here is what I did: (**ensure your are in ml-stable-diffusion dir**)

## 1. Use the specific git hash branch in `apple/ml-stable-diffusion`

```
git checkout c891f4357b2269853152d774d97327faab746e7c
```

## 2. create a new conda env based on `python 3.10.10` and activate it

```
conda create -n coreml_stable_diffusion python=3.10.10 -y

conda activate coreml_stable_diffusion
```

## 3. Use this pinned dependencies instead of the original dependencies in `apple/ml-stable-diffusion/requirements.txt`

accelerate==0.24.1
antlr4-python3-runtime==4.9.3
argmaxtools==0.1.13
attrs==24.2.0
beartype==0.18.5
cattrs==23.2.3
certifi==2024.7.4
charset-normalizer==3.3.2
contourpy==1.2.1
coremltools==7.1
cycler==0.12.1
diffusers==0.23.1
diffusionkit==0.2.16
exceptiongroup==1.2.2
filelock==3.15.4
fonttools==4.53.1
fsspec==2024.6.1
huggingface-hub==0.17.3
idna==3.7
importlib_metadata==8.2.0
iniconfig==2.0.0
invisible-watermark==0.2.0
jaxtyping==0.2.33
Jinja2==3.1.4
joblib==1.4.2
kiwisolver==1.4.5
MarkupSafe==2.1.5
matplotlib==3.9.1.post1
mlx==0.16.2
mpmath==1.3.0
networkx==3.3
numpy==1.23.5
omegaconf==2.3.0
opencv-python==4.10.0.84
packaging==24.1
pillow==10.4.0
pluggy==1.5.0
protobuf==3.20.3
psutil==6.0.0
pyaml==24.7.0
pyparsing==3.1.2
pytest==8.3.2
python-dateutil==2.9.0.post0
-e git+https://github.com/apple/ml-stable-diffusion.git@c891f4357b2269853152d774d97327faab746e7c#egg=python_coreml_stable_diffusion
PyWavelets==1.6.0
PyYAML==6.0.2
regex==2024.7.24
requests==2.32.3
safetensors==0.4.4
scikit-learn==1.1.2
scipy==1.14.0
sentencepiece==0.2.0
six==1.16.0
sympy==1.13.1
tabulate==0.9.0
threadpoolctl==3.5.0
tokenizers==0.14.1
tomli==2.0.1
torch==2.1.0
tqdm==4.66.5
transformers==4.34.1
typeguard==2.13.3
typing_extensions==4.12.2
urllib3==2.2.2
zipp==3.19.2

## 4. install the requirements from `requirements.txt` rather than  `conda install -e .`

```
python install -r requirement.txt
```

## 5. continue your conversion workflow
